### ISSUE TITLE
Broken second-order floating dialogs

### ISSUE THREAD

**Issue by shader** [ch11ng/exwm#554]
When I click a button on a floating dialog that opens _another_ floating dialog, EXWM ends up in an unstable state. For example, when downloading a file with Firefox and selecting the "open with..." option, that pops up a program selector. The other place I've had the problem is selecting a file to save a torrent being added to Deluge. The "Add Torrent" dialog is floating, and it wants to open another floating path selector.

Usually the 2nd dialog will not render properly, instead showing the background image from my login screen. Sometimes I have been able to close it, but other times the application can close but the buffer and hole in my screen stick around.

Any suggestions for how to debug, or configuration I might have gotten wrong?

**Comment by ch11ng** [comment]
Could you provide your EXWM-related configs? I tried your recipe with Firefox and was not able to reproduce the problem.

BTW if you don't want floating X windows at all you can simply set `exwm-manage-force-tiling` to `t`.

**Comment by shader** [comment]
Ok, finally had a chance to spend some time debugging this. I still don't know what's causing it, but I have some more information. Both this and the other issue with the split dialogs while viewing split windows happen to me only when I load my full init file during startup.

First I commented out everything after my exwm config in my init.el file, and then tried to progressively evaluate it to find the problem. Even after evaluating my whole init file, I didn't reproduce either bug. However, restarting again with everything at once reproduced both. I tried again, splitting everything after the exwm configs into a separate file. If I comment out `(require 'main)` from the init.el file, and then manually load it later, I get all of my configs and no issues with exwm. If I uncomment the line and try to load everything during startup, I get both bugs.

Any suggestions? I guess the next step is to progressively move things into "hard loaded" code, and keep reloading until I get the bug... Sounds like a lot of work though, so if you can give me any hints I'd be happy to try that first.

**Comment by shader** [comment]
Ok, I finally figured out what the problem is. I was able to narrow down the source of the problem in my emacs config to the offending line in the following snippet.

```elisp
(use-package persp-mode
  :config
  (setq persp-autokill-buffer-on-remove 'kill-weak)
  :hook after-init  ;; <--- offending line
  :bind-keymap
  ("C-c p" . persp-key-map))
```

Removing this line made both the split windows and second-order dialogs problem go away.

On further testing, it seems that this made it work because EXWM conflicts somehow with persp-mode, and by either not having the hook line, or evaluating the form after the after-init-hook has already been called, persp-mode is not enabled. If I manually activate persp-mode, the problem recurrs. Most importantly, turning _off_ persp-mode makes it go away again.

For now, I'll go without using persp-mode, but I'll create an issue on both projects to track the bug.
